src/dataflow/src/render/mod.rs

```diff
@@ -813,6 +813,33 @@ where
                 self.collections.insert(relation_expr.clone(), (oks, err));
                 true
             }
+
+            RelationExpr::Join {
```

Comment:
the comment on line 742/743 about the Filter on join special case looks like its no longer true?

Reply:
I think that got fixed in the commit just above (2bdd0bb). Probably landed as you were reading, but I think it addresses this.

Reply:
yup something something race conditions :)